src/Api/EmptyCart.php

```diff
@@ -0,0 +1,97 @@
+<?php namespace Lean\Woocommerce\Api;
+
+use Lean\AbstractEndpoint;
+use Epoch2\HttpCodes;
+use Lean\Woocommerce\Utils\ErrorCodes;
+use Lean\Woocommerce\Controllers\UserController;
+
+/**
+ * Class Cart.
+ *
+ * @package Leean\Woocomerce\Modules\Cart
+ */
+class EmptyCart extends AbstractEndpoint {
+	/**
+	 * Endpoint path
+	 *
+	 * @Override
+	 * @var String
+	 */
+	protected $endpoint = '/ecommerce/empty_cart';
+
+	/**
+	 * Endpoint callback override.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 *
+	 * @return array|\WP_Error
+	 */
+	public function endpoint_callback( \WP_REST_Request $request ) {
+		$method = $request->get_method();
+
+		if ( \WP_REST_Server::CREATABLE === $method ) {
+			return self::empty_cart( $request );
+		} else {
+			return new \WP_Error(
+				ErrorCodes::METHOD_ERROR,
+				'Method not allowed',
+				[ 'status' => HttpCodes::HTTP_METHOD_NOT_ALLOWED ]
+			);
+		}
+	}
+
+	/**
+	 * Set the options of the endpoint. Allow http methods.
+	 *
+	 * @return array
+	 */
+	protected function endpoint_options() {
+		return [
+			'methods' => [ \WP_REST_Server::CREATABLE ],
+			'callback' => [ $this, 'endpoint_callback' ],
+			'args' => $this->endpoint_args(),
+		];
+	}
+
+	/**
+	 * Set the args for this endpoint.
+	 *
+	 * @return array
+	 */
+	public function endpoint_args() {
+		return [
+			'token_id' => [
+				'default' => false,
+				'required' => false,
+				'validate_callback' => function( $token_id ) {
+					return false === $token_id || is_string( $token_id );
```

Comment:
This is only a validation. 

Because default value for token_id is false (in case we don't specify a token), that means if token_id === false, it's correctly validated.   So we are ok either if is a string, or if it's the default value (false).

How does it sound for u? @mitogh  Thank you!